### ISSUE TITLE
Use phly/http for HTTP message implementations

### ISSUE THREAD

**Issue by kasperg** [egeloen/ivory-http-adapter#74]
Thank you for sharing your work. It appears to be just what I was looking for for a recent project.

I noticed that the project provides [its own implementations of the PSR-7 interfaces](https://github.com/egeloen/ivory-http-adapter/tree/master/tests/Message). Matthew Weier O'Phinney who leads PSR-7 released [phly/http](https://github.com/phly/http) which also provides implementations of these interfaces.

I was thinking that this project could pull in his work instead on using its own implementations?

As I see it the client adapters are the core of this project - not the message implementations. This change would reduce the amount to code to maintain. Also given his position I would say that we can expect the package to be up to date with the latest changes to PSR-7 sooner than this project can implement it on its own - see #68.

**Comment by egeloen** [comment]
:+1: This is something I'm already thinking about :)

**Comment by sagikazarmark** [comment]
I am not sure whether that package is rather an experiment or not. The author should be asked about it whether it is intended to be used in production or not.

**Comment by egeloen** [comment]
Yeah, after looking at it, it is only compatible with PHP >= 5.4.8 according to its composer.json The, i'm not sure to use it...

**Comment by jeromegamez** [comment]
Are you sure you want to continue to support PHP ~5.3? After all, it's deprecated, and for PHP 5.4, people shouldn't use outdated versions anyway - and according to http://blog.ircmaxell.com/2014/12/php-install-statistics.html , the majority of PHP installations are newer than PHP 5.4.8.

My comments are not just aimed at phly/http, but a thought in general...

**Comment by egeloen** [comment]
If everyone agree for bumping PHP to 5.4.8+, Im :+1: too!

**Comment by toin0u** [comment]
:+1:

**Comment by kasperg** [comment]
:+1: here also if that will help bring the project forward.

Sidenote: If we do this perhaps it is also worth looking into if the project can use [InterNations/http-mock](https://github.com/InterNations/http-mock) to test against the PHP webserver. [The current setup](https://github.com/egeloen/ivory-http-adapter/blob/master/CONTRIBUTING.md#bug-fix) seems rather complex.

**Comment by weierophinney** [comment]
I built phly/http to test out PSR-7, and, once I took over as editor of the proposal, to track any changes introduced. I test each release, so in terms of code robustness, it's in good shape. However, the proposal has changed quite a bit in the last four months, which is why I've not marked a stable version; each minor version has introduced backwards breaking changes!

All that said, I've got one PR about ready to push against PSR-7, and then anticipate being able to put it up to review, which is the last step before voting, and the first point at which the API freezes. I do not anticipate we'll see too many changes from this point forward.
